analyses/hotspots-detection/01-create-hotspot-maf.Rmd

```diff
@@ -0,0 +1,246 @@
+  
+  # join vardict
+  full_join(
+    filter(combined_maf_hotspots_filt,caller=="vardict"),
+    by=setdiff(names(combined_maf_hotspots_filt ), unique_cols)) %>%
+  # rename *_count columns to vardict_*count
+  rename_at(vars(unique_cols), ~str_c("vardict_", .))
+
+
+# We will select unique cols for each variant call in the order 
+# strelka2 > mutect2 > lancet > vardict
+combined_maf_hotspots_filt <- combined_maf_hotspots_filt %>%
+  # Given a set of vectors, coalesce() finds the first non-missing value at each position. 
+  mutate( "t_depth" = coalesce(strelka2_t_depth,mutect2_t_depth,lancet_t_depth,vardict_t_depth),
+          "n_depth" = coalesce(strelka2_n_depth,mutect2_n_depth,lancet_n_depth,vardict_n_depth),
+          "t_ref_count" = coalesce(strelka2_t_ref_count,mutect2_t_ref_count,lancet_t_ref_count,vardict_t_ref_count),
+          "n_ref_count" = coalesce(strelka2_n_ref_count,mutect2_n_ref_count,lancet_n_ref_count,vardict_n_ref_count),
+          "t_alt_count" = coalesce(strelka2_t_alt_count,mutect2_t_alt_count,lancet_t_alt_count,vardict_t_alt_count),
+          "n_alt_count" = coalesce(strelka2_n_alt_count,mutect2_n_alt_count,lancet_n_alt_count,vardict_n_alt_count),
+          "vcf_qual" = coalesce(strelka2_vcf_qual,mutect2_vcf_qual,lancet_vcf_qual,vardict_vcf_qual),
+          # calculate VAF
+          VAF = t_alt_count/(t_alt_count+t_ref_count))%>%
+  # Filter with vaf_filter parm in this notebook
+  dplyr::filter(VAF >= params$vaf_filter) 
+
+
+```
+
+## Identify read support for variant type per caller
+
+We will try to identify the read support for each variant type from each callers to visualize if any callers call hostspot sites in specific variant types with low read support calls 
+```{r}
+
+## adding in caller info to plot caller information
+combined_maf_hotspots_filt <- combined_maf_hotspots_filt %>%
+  left_join(caller_counts,by=join_cols) 
+
+ggplot(combined_maf_hotspots_filt[,c("caller","caller_count","t_alt_count")],
+       aes(x=as.factor(caller_count),y=t_alt_count))+
+  geom_violin()+ 
+  ggpubr::stat_compare_means()
+
+```
+Hotspot sites called by just 1 caller seem to have majority low read support and a few are found in 3 caller ( now captured in 3 callers because of the variant being present in vardict)
+
+## Identify calls unique to a caller
+Lets look at these calls that are unique to 1 caller.
+
+```{r}
+
+unique_1caller_hotspots <- combined_maf_hotspots_filt %>%
+  filter(caller_count==1)
+
+unique_1caller_hotspots
+```
+There are 79 SNV or indel calls are only called by 1 caller.
+
+Vardict calls a lot of variants summarized [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/README.md)
+In snv-callers consensus calling method we didn't consider vardict calls so we will need to filter out vardict only calls from the hotspots to
+be consistent 
+```{r}
+ combined_maf_hotspots_filt <- combined_maf_hotspots_filt %>% filter(caller !="vardict")
+```
```

Comment:
Now that we have to gather all the hotspots calls in a maf, this step https://github.com/kgaonkar6/OpenPBTA-analysis/blob/ab9c3b2ba7f093c3e1254093fc92ac01d83ed28b/analyses/hotspots-detection/01-create-hotspot-maf.Rmd#L130-L169 takes a lot of RAM so I'm running in m5 instance. I'll  report back after a rerun about the vardict -only calls and updated file.

Reply:
The above RAM issue was because if had multiple rows in *RDS files that were being merged in this script. I have added unique() to each output from filterMaf.R which as solved the issue and now can be run locally without much RAM requirements.


The 79 calls unique to 1 caller was from mafs in v18 release. In v19 , 148 calls are unique to 1 caller and this is the distribution of calls that are unique to just 1 caller:
caller<chr> | n<int>
-- | --
lancet | 38
mutect2 | 24
strelka2 | 9
vardict | 77

[unique_to_1_caller_hotspots.txt](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/6467210/unique_to_1_caller_hotspots.txt)

Reply:
This is the distribution of calls unique to 1 caller : 
![Rplot02](https://user-images.githubusercontent.com/34580719/118004232-464cf380-b317-11eb-92c8-bd9b65dcbe3f.png)

Reply:
@kgaonkar6 can you check whether the HLA-A calls are from one caller or multiple?

>I think I asked this before, but if implement a consistent rule that we require >= 2 callers with no real loss, that seems simpler & more consistent with what I have seen elsewhere.

@jashapiro given these comparisons, I wonder if we should implement a >=2 callers if vardict is is included or a >=1 caller if it is from one of the other 3?

Reply:
Another option could be adding all of these and having a column for caller, such that users can filter out if just from VarDict, for example.

Reply:
I dug into HLA-A, because it seemed spurious to me. It looks like the mutations were called by both Mutect2 and Lancet, and for each sample, there are two adjacent mutations. Turns out, this is an [MNV](https://www.ncbi.nlm.nih.gov/snp/?term=rs796080364) in dbSNP. It is benign in PolyPhen, so I am wondering if this should even be a real hotspot.

Reply:
I've added all the calls that are unique to 1 caller in the notebook for review as suggested by @jharenza